File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/RotationalCipher.cs
using System.Text;

namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        var cipher = new StringBuilder(text.Length);
        foreach (var character in text)
        {
            if (char.IsLetter(character))
            {
                var offset = char.IsUpper(character) ? 'A' : 'a';
                cipher.Append((char) ((character + shiftKey - offset) % 26 + offset));
            }
            else
            {
                cipher.Append(character);
            }
        }

        return cipher.ToString();    
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/RotationalCipher.java
package exercism;

public class RotationalCipher {
    private final int shiftKey;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        var result = new StringBuilder();
        for (char character : data.toCharArray()) {
            if (Character.isAlphabetic(character)) {
                var base = Character.isLowerCase(character) ? 'a' : 'A';
                character = (char) ((character - base + shiftKey) % 26 + base);
            }
            result.append(character);
        }
        return result.toString();
    }
}


